### PR TITLE
Data block allocator

### PR DESCRIPTION
**Pull request by VenkyOS**

### CODE REVIEW

File: doc/HLD_Data_Block_Allocator.rst [hunk 1]
```diff
@@ -0,0 +1,407 @@
+=====================
+Data Block Allocator
+=====================
+
+This document presents a high level design (HLD) of a data block allocator for Motr C2 core. The main purposes of this document are: (i) to be inspected by C2 architects and peer designers to ascertain that high level design is aligned with C2 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
```
Comment (huanghua78) [inline doc/HLD_Data_Block_Allocator.rst:5]: Remove the "C2".
If "C2" appears alone, replace it with Motr.

File: doc/HLD_Data_Block_Allocator.rst [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+
+- Object-based pre-allocation and group-based pre-allocation are both supported.
+
+- Blocks are allocated in a bunch. Multi-blocks can be allocated at one time. Buddy algorithm is used to do allocation.
+
+***************************
+Functional Specification
+***************************
+
+C2_mb_format
```
Comment (huanghua78) [inline doc/HLD_Data_Block_Allocator.rst:75]: Relace the "C2_" prefix with "m0_" prefix.

File: doc/HLD_Data_Block_Allocator.rst (hunk 1 above)
Comment (VenkyOS) [reply doc/HLD_Data_Block_Allocator.rst]: Incorporated.

File: doc/HLD_Data_Block_Allocator.rst (hunk 2 above)
Comment (VenkyOS) [reply doc/HLD_Data_Block_Allocator.rst]: Incorporated.

**Review by huanghua78 — APPROVED** [review]

**Review by pujamudaliar — APPROVED** [review]

**Pull request merged**